nrf5/hal/hal_temp.c

```diff
@@ -25,18 +25,36 @@
  */
 
 #include <stdio.h>
+#include <stdint.h>
 #include "mphalport.h"
 #include "hal_temp.h"
 
+#if BLUETOOTH_SD
```

Comment:
You are right, that could have worked. Until now i have not interfaced the sd_* other than through the ble_drv driver, trying to abstract it. Similar #ifdef is also present in hal_irq for nvic access.